src/babel.js

```diff
@@ -155,7 +163,25 @@ module.exports = () => {
     },
     visitor: {
       CallExpression(p, { file, opts }) {
-        if (proceed) return
+        const { filename } = file.opts
+
+        if (proceed && opts.setHash) {
+          const names = Object.keys(dataFromComment)
+          for (let i = 0; i < names.length; i++) {
+            opts.setHash({
+              id: dataFromComment[names[i]].id,
+              name: names[i],
+              filename,
+              loc: dataFromComment[names[i]].loc,
+            })
+          }
+
+          return
+        }
+
+        if (proceed) {
+          return
+        }
```

Comment:
Would mind to make it a bit simpler maybe?

```suggestion
        const { filename } = file.opts

        if (proceed) {
          if (opts.setHash) {
            Object.entries(dataFromComment).forEach(([name, data])=> {
              opts.setHash({
                filename,
                name,
                id: data.id,
                loc: data.loc,
              })
            })
          }
        
          return;
        }
```